astra/src/main/java/com/slack/astra/metadata/core/EtcdPartitioningMetadataStore.java

```diff
@@ -0,0 +1,236 @@
+package com.slack.astra.metadata.core;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.io.Closeable;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import org.apache.zookeeper.data.Stat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * EtcdPartitioningMetadataStore is a class which provides consistent Etcd apis for partitioned
+ * metadata store operations.
+ *
+ * <p>Every method provides an async and a sync API. In general, use the async API you are
+ * performing batch operations and a sync if you are performing a synchronous operation on a node.
+ *
+ * <p>This class is the Etcd counterpart to ZookeeperPartitioningMetadataStore. It is designed to be
+ * used with AstraPartitioningMetadataStore for migrating between Zookeeper and Etcd.
+ */
+public class EtcdPartitioningMetadataStore<T extends AstraPartitionedMetadata>
+    implements Closeable {
+  private static final Logger LOG = LoggerFactory.getLogger(EtcdPartitioningMetadataStore.class);
+  protected final String storeFolder;
+
+  private final MeterRegistry meterRegistry;
+
+  private final String ASTRA_ETCD_PARTITIONING_CREATE_CALL = "astra_etcd_partitioning_create_call";
+  private final String ASTRA_ETCD_PARTITIONING_DELETE_CALL = "astra_etcd_partitioning_delete_call";
+  private final String ASTRA_ETCD_PARTITIONING_LIST_CALL = "astra_etcd_partitioning_list_call";
+  private final String ASTRA_ETCD_PARTITIONING_GET_CALL = "astra_etcd_partitioning_get_call";
+  private final String ASTRA_ETCD_PARTITIONING_UPDATE_CALL = "astra_etcd_partitioning_update_call";
+  private final String ASTRA_ETCD_PARTITIONING_ADDED_LISTENER =
+      "astra_etcd_partitioning_added_listener";
+  private final String ASTRA_ETCD_PARTITIONING_REMOVED_LISTENER =
+      "astra_etcd_partitioning_removed_listener";
+  private final String ASTRA_ETCD_PARTITIONING_CACHE_INIT_HANDLER_FIRED =
+      "astra_etcd_partitioning_cache_init_handler_fired";
```

Comment:
For ZK, I believe we have only one set of these and just label it with the store. I don't think we need separate metrics

Reply:
I meant that ZK has one set of metrics for both stores and partitioned stores, so we probably only need one set for etcd, instead of separate partitioned ones